TrainingLoop/Callbacks/CSVLogger.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import ModelSupport
+
+/// A callback-based handler for logging the statistics to CSV file.
+public class CSVLogger {
+  public var path: String
+
+  let foundationFS: FoundationFileSystem
+  let foundationFile: FoundationFile
+
+  /// Create an instance that log statistics during the training loop.
```

Comment:
“during the training loop” is not true about this initializer, only about how you happen to use it later.  The doc comment should say something about how the parameters are used.

```suggestion
  /// Creates an instance that logs to a file with the given path.
```

Reply:
Done.